notebooks/python/raw/ex_2.ipynb

```diff
@@ -133,6 +181,34 @@
     "q3.check()"
    ]
   },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#%%RM_IF(PROD)%%\n",
+    "def to_smash(total_candies, n_friends=3):\n",
+    "    return n_friends % total_candies\n",
+    "\n",
+    "q3.check()\n",
+    "q3.assert_failed()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#%%RM_IF(PROD)%%\n",
+    "def to_smash(total_candies, n_friends=3):\n",
+    "    return total_candies % n_friends\n",
+    "\n",
+    "q3.check()\n",
+    "q3.assert_passed()"
```

Comment:
Why not combine `check()` and `assert_passed()` into a single method `assert_check_passes()`.  

I don't think we're going to call `assert_passed` without calling `check` first.

Reply:
I considered that, but ended up mildly preferring the clarity of this approach. It makes it clear that we're exercising the normal `.check()` code path, including seeing the exact output the user would get from calling `q3.check()` with this solution, and then also asserting that this solution passed our tests. (Rather than just doing the latter).

FWIW, with copy-paste taken into account, I think both approaches involve about the same amount of typing.

Reply:
I'm less concerned about the extra typing than I am that the raw notebooks are getting pretty noisy, with a lot of testing code that's not shown to the user (and which distracts the author from what we show to the user, especially when authoring on a laptop). So my hope is to save one line of code in each of these testing cells... and I think slightly obscuring the fact this calls `check` is worth it.